src/App.tsx

```diff
@@ -1,35 +1,44 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { Component } from 'react';
+import SearchGroup from './components/SearchGroup';
+import './App.css';
+import { Character } from './types';
+import CharacterCardList from './components/CharacterCardList';
+import BtnThrowError from './components/BtnThrowError';
+import ErrorBoundary from './components/ErrorBoundary';
 
-function App() {
-  const [count, setCount] = useState(0)
+interface AppState {
+  characters: Character[];
+  isLoading: boolean;
+}
+
+class App extends Component<object, AppState> {
+  updateCharacters = (data: Character[], isLoading = false) => {
+    this.setState({ characters: data });
+    this.setState({ isLoading: isLoading });
+  };
```

Comment:
this can be added as one object
this.setState({ characters: data, isLoading: isLoading });